### PR TITLE
IRCv3 Message Tag Support

### PR DESCRIPTION
**Pull request by lcycon**
This adds support for parsing [IRCv3 Message Tags](http://ircv3.net/specs/core/message-tags-3.2.html) into the IrcMessage record.

### CODE REVIEW

File: stack.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# This file was automatically generated by 'stack init'
```
Comment (dom96) [inline stack.yaml]: hrm, what is this Stack file? (I'm not really doing much Haskell dev these days)

Comment (lcycon) [inline stack.yaml]: [Stack](https://www.stackage.org/) is a project that's tried to solve the "cabal hell" issue by curating known-working package lists. In practice, this yaml file just serves to tell the stack executable what versions of the compiler and whatnot to use.

I take it as a big confidence boost in the "compilability" of a project I am looking at using, but I can remove this all from the PR if you'd like

Comment (ehamberg) [inline stack.yaml]: Yeah, adding `stack.yaml` doesn't seem to belong in a PR whose topic is IRCv3 message tag support and should probably be a separate PR. :)

Comment (lcycon) [inline stack.yaml]: I see so many of these nowadays, it's almost mechanical to add them to a project :P

I'll break it out into a separate PR

**Comment by lcycon** [comment]
Updated to exclude Stackage changes
